pkg/asset/ignition/machine/node.go

```diff
@@ -13,6 +13,12 @@ import (
 // pointerIgnitionConfig generates a config which references the remote config
 // served by the machine config server.
 func pointerIgnitionConfig(installConfig *types.InstallConfig, rootCA []byte, role string) *ignition.Config {
+	ignitionHost := fmt.Sprintf("api-int.%s:22623", installConfig.ClusterDomain())
+	if installConfig.Platform.BareMetal != nil {
```

Comment:
I found a couple of other instances of platform checks in common code so I'm hoping this is okay. I'm open to other suggestions though.